graphbolt/src/cuda/neighbor_sampler.cu

```diff
@@ -528,18 +534,107 @@ c10::intrusive_ptr<sampling::FusedSampledSubgraph> SampleNeighbors(
         }
       }));
 
-  // Convert output_indptr back to homo by discarding intermediate offsets.
-  output_indptr =
-      output_indptr.slice(0, 0, output_indptr.size(0), fanouts.size());
+  torch::optional<torch::Tensor> edge_offsets = torch::nullopt;
```

Comment:
Add step by step comments to help people understand the code?

Reply:
Added comments, hopefully it is a bit more clear.